packages/core/strapi/lib/commands/builders/tasks/dts.js

```diff
@@ -0,0 +1,191 @@
+'use strict';
+
+const path = require('path');
+const chalk = require('chalk');
+const ora = require('ora');
+const ts = require('typescript');
+
+/**
+ * @description Load a tsconfig.json file and return the parsed config
+ *
+ * @internal
+ *
+ * @type {(args: { cwd: string; path: string }) => Promise<ts.ParsedCommandLine>)}
+ */
+const loadTsConfig = async ({ cwd, path }) => {
+  const configPath = ts.findConfigFile(cwd, ts.sys.fileExists, path);
```

Comment:
Note: In this situation

```
/
  tsconfig.dist.json
  / my-dir
    / my project
      <I AM HERE> -> findConfigFile(...)
```

TypeScript will return the root tsconfig.

I don't know if this is the wanted behavior, just wanted to point it out in case you weren't aware

Reply:
I think it's probably better to traverse upwards and find things, but thanks for making me aware ✨